### PR TITLE
Webc 1626 live decorator

### PR DESCRIPTION
**Pull request by eitanavgil**

### CODE REVIEW

File: src/decorator/live-decorator.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { KalturaLivePlugin } from "../plugin";
```
Comment (eitanavgil) [inline src/decorator/live-decorator.ts]: @esakal  this is a naive implementation that does not really do anything. The implementation of logic will be done by other tickets

File: src/plugin.tsx [hunk 2]
```diff
@@ -30,54 +31,82 @@ interface KalturaLivePluginConfig {
...
+    // value of last 'isLive' API call
+    public get broadcastState(): LiveBroadcastStates {
+        return this._broadcastState;
+    }
+
+    private _isEntryLiveType = () => {
+        const { playerConfig } = this._configs;
+        if (playerConfig && playerConfig.sources.type === "Live") {
+            this._isLiveEntry = true;
+            this._checkIsLive();
```
Comment (esakal) [inline src/plugin.tsx]: the names here are a bit confusing. usually method that starts with `is...` doesn't actually trigger any logic it only query existing state.

File: src/plugin.tsx [hunk 3]
```diff
@@ -30,54 +31,82 @@ interface KalturaLivePluginConfig {
...
+        }
     };
 
-    private _checkIsLive = (id: string) => {
+    private _checkIsLive = () => {
```
Comment (esakal) [inline src/plugin.tsx]: same here, the names here are a bit confusing. consider `updateLiveStatus`

File: src/plugin.tsx [hunk 4]
```diff
@@ -30,54 +31,82 @@ interface KalturaLivePluginConfig {
...
         const { pluginConfig } = this._configs;
-        const protocol = KalturaPlaybackProtocol.auto;
+        const protocol = KalturaPlaybackProtocol.hls;
+        const id = this._player.config.sources.id; // todo - consider do this once on  media load
```
Comment (esakal) [inline src/plugin.tsx]: consider guarding / handling scenarios when user calls check is live before you actually have entry id

File: src/plugin.tsx [hunk 5]
```diff
@@ -105,13 +136,21 @@ export class KalturaLivePlugin implements OnMediaUnload, OnRegisterUI, OnMediaLo
                 }
             }
         );
+        // interval - todo handle clear and cancel
+        setTimeout(this._checkIsLive, 10000);
```
Comment (esakal) [inline src/plugin.tsx]: consider getting the `10000` from configuration and default to that value

File: src/plugin.tsx (hunk 4 above)
Comment (eitanavgil) [reply src/plugin.tsx]: @esakal  
This is not a possible scenario. _checkIsLive is called after we know this is a live entry - hence we have an entry!

**Pull request merged**